### DISCUSSION TITLE
Livewire requests completely failing inside iframes/embeds in Safari

### DISCUSSION THREAD

**Discussion by 7ammer** [livewire/livewire#5329] [7. Bugs (archived)]
Livewire requests completely fail inside an iframe in safari.

I have tried setting session.php -> `same_site = 'none'` and made sure session.php -> `secure = true`. This fixes issues on other browsers like chrome and firefox. However, unless you disable privacy -> `prevent cross-site tracking` in Safari preferences livewire just fails to update in any way.

This makes it impossible to use livewire for embedded content :(

I believe the problem is that safari is blocking all third party cookies by default. Unfortunately this completely brakes laravel's sessions and crsf prevention which then also brakes Livewire.
https://webkit.org/blog/10218/full-third-party-cookie-blocking-and-more/

Does anyone have any helpful advice here because I'm out of ideas.

**Comment by joshhanley** [comment]
@7ammer is that just unique to Livewire or does Laravel not work either due to the blocked third party cookies?

**Comment by 7ammer** [comment]
Hi @joshhanley,

Both laravel and livewire fail when laravel cookie sessions or csrf tokens are used in the scenario I've mentioned.

As livewire uses csrf tokens it is also failing and returns a 419 unknown status for every request.

To work around this issue I could exclude the relevant livewire endpoints from validating the csrf token like so:

```
    protected $except = [
        '/livewire/message/counter',
        '/livewire/message/modal',
    ];
```

However, this feels pretty insecure.

**Reply by joshhanley** [reply]
@7ammer seeing as this is more of an underlying Laravel issue (keep in mind Livewire is built on top of it), I would suggest seeing if anyone on the Laravel forums has a work around for this issue.

Yeah excluding the routes from the CSRF check wouldn't be a good idea in my opinion.

Hope this helps!
